src/documentation.lisp

```diff
@@ -14,6 +14,11 @@
        (error "The type of documentation `~S' is not a symbol." type))
      (oget x "vardoc"))))
 
+(defun function-lambda-list (function)
+  "Return the lambda-list of FUNCTION."
+  (let ((lambda-list (oget (fdefinition function) "lambdalist")))
+    (when lambda-list
+      (read-from-string lambda-list))))
```

Comment:
As mentioned in the comments, this feels a bit weird.
We could just dump the lambda-list with `literal` from the compiler I guess so there is no need to re-read it?

Reply:
Ok. I dumped it as a string. I'm not expert on the compiler internals.